packages/block-editor/src/components/inserter/style.scss

```diff
@@ -14,43 +15,107 @@ $block-inserter-search-height: 38px;
 	@include break-medium {
 		position: relative;
 	}
-}
 
-.block-editor-inserter__popover > .components-popover__content {
-	@include break-medium {
-		overflow-y: visible;
-		height: $block-inserter-content-height + $block-inserter-tabs-height + $block-inserter-search-height;
+	.block-editor-inserter__toggle {
+		display: inline-flex;
+		align-items: center;
+		color: $dark-gray-500;
+		background: none;
+		cursor: pointer;
+		border: none;
+		outline: none;
+		transition: color 0.2s ease;
+		@include reduce-motion("transition");
 	}
 }
 
-.block-editor-inserter__toggle {
-	transition: color 0.2s ease;
-	@include reduce-motion("transition");
+// Popover container.
+.block-editor-inserter__popover {
+
+	// The in-canvas appender library.
+	> .components-popover__content {
+		height: 440px;
+	}
+
+	// The toolbar library.
+	&.is-top-toolbar-inserter > .components-popover__content {
+		@include break-medium {
+			overflow-y: visible;
+			height: 92vh; // This is limited by a max-height from the popover component itself.
+			width: 82vw; // This is limited by a max-width.
```

Comment:
I don't like that this is using random numbers, this ignores the fact that the inserter is a reusable component that can be used anywhere. 

Can we just remove these, have a max-width and height 100% (popover should adapt it according to the available space)?

Reply:
This is a good thought, and is related to both https://github.com/WordPress/gutenberg/pull/19836#discussion_r377496433 and https://github.com/WordPress/gutenberg/pull/19836#discussion_r377494846.

The challenge is, we need for this to be responsive, i.e. fill the full height of the screen, and at least a big chunk of the screen horizontally at many breakpoints.

Absolute positioning can be used for that normally. But in this case, the popover container acts as a _relative positioner_, meaning that the coordinates, the width and the height of any child element is relative to that container. And it's 0 pixels wide and tall, which in this case was obviously solved with viewport units. 

I'm afraid a refactor of this is probably beyond my skill to address.